### PR TITLE
MIPS n32 ABI support, MSA support detection and rename ARCH, ARCHFLAGS

### PR DESCRIPTION
**Pull request by sva-img**
MIPS n32 ABI support, MSA support detection and rename ARCH, ARCHFLAGS

Signed-off-by: Shivraj Patil <shivraj.patil@imgtec.com>

### CODE REVIEW

**Comment by xianyi** [comment]
Hi @sva-img ，

Why do you need to  rename ARCH, ARCHFLAGS?

**Comment by sva-img** [comment]
Hi xianyi,

Existing ARCH and ARCHFLAGS is confusing whether it means architecture flags or archiever flags. 
I think, renaming ARCH to AR and ARCHFLAGS to ARFLAGS makes it more clear that these are archiever only flags.

If this looks good, I will resubmit this change in new PR.

File: kernel/Makefile [hunk 1]
```diff
@@ -4,7 +4,6 @@ endif
 
 TOPDIR	= ..
 include $(TOPDIR)/Makefile.system
-include $(TOPDIR)/Makefile.conf
```
Comment (xianyi) [inline kernel/Makefile]: @sva-img , why do you delete Makefile.conf?

**Comment by xianyi** [comment]
@sva-img ,  `ARCH` and `ARCHFLAGS` flags come from upstream LAPACK implementation. Thus,   I suggest keep the flags with upstream.

File: kernel/Makefile (hunk 1 above)
Comment (sva-img) [inline kernel/Makefile]: @xianyi, This is reverted to original code which was added in previous commit.

**Comment by sva-img** [comment]
Ok. I have reverted rename change to original in last commit.

**Comment by sva-img** [comment]
@xianyi, Please have a look of combined changes from 'all commits' for better clarity.

**Pull request merged**